### PR TITLE
Report protocol in request via header

### PR DESCRIPTION
**Pull request by shukitchan**
See #1506

### CODE REVIEW

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/freebsd-github/1656/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/linux-github/1552/ for details.

**Comment by atsci** [comment]
Intel CC build *successful*! See https://ci.trafficserver.apache.org/job/icc-github/88/ for details.

**Comment by atsci** [comment]
clang-analyzer build *successful*! See https://ci.trafficserver.apache.org/job/clang-analyzer-github/220/ for details.

File: proxy/http/HttpTransactHeaders.cc [hunk 1]
```diff
@@ -741,26 +741,14 @@ HttpTransactHeaders::insert_via_header_in_request(HttpTransact::State *s, HTTPHd
...
-    via_string += 5;
-  } else {
-    *via_string++ = '/';
-    *via_string++ = '0' + HTTP_MAJOR(hversion);
-    *via_string++ = '.';
-    *via_string++ = '0' + HTTP_MINOR(hversion);
+  char const *proto_buf[10]; // 10 seems like a reasonable number of protos to print
+  int retval = s->state_machine->populate_client_protocol(proto_buf, countof(proto_buf));
+  for (int i = 0; i < retval; i++) {
+    memcpy(via_string, proto_buf[i], strlen(proto_buf[i]));
```
Comment (zwoop) [inline proxy/http/HttpTransactHeaders.cc]: Aaah this is super unfortunate IMO, so now we have to do strlen() on all these strings, even though they are static? :-/. I remember having this conversation about the public APIs, but can we make something internal that doesn't have this shortcoming?

Comment (SolidWallOfCode) [inline proxy/http/HttpTransactHeaders.cc]: You mean like `StringView`? E. g.
```
std::array<StringView,10> proto_buf;
int retval = s->state_machine->populate_client_protocol(proto_buf, proto_buf.size());
for ( StringView const& s : proto_buf ) memcpy(via_string, s.ptr(), s.size());
```

Comment (zwoop) [inline proxy/http/HttpTransactHeaders.cc]: I mean exactly like that.

Comment (maskit) [inline proxy/http/HttpTransactHeaders.cc]: I'm fine with use of StringView, but I think we should use it both request side and response side. Why don't we  do the change separately from this FIX? Ideally, the logic should be unified.

Comment (shukitchan) [inline proxy/http/HttpTransactHeaders.cc]: it is ok. I can make the change tonight for both request and response in this PR.

Comment (shukitchan) [inline proxy/http/HttpTransactHeaders.cc]: Run into some trouble and Alan is helping me out now.

**Comment by atsci** [comment]
Linux build *failed*! See https://ci.trafficserver.apache.org/job/RAT-github/4/ for details.

**Comment by shukitchan** [comment]
#1534 is merged. So closing this one. Thanks @SolidWallOfCode

**Pull request closed without merging**